src/tx-format/transaction.md

```diff
@@ -133,16 +133,15 @@ Creates a contract with contract ID as computed [here](../identifiers/contract-i
 The transaction is created by the block producer and is not signed. Since it is not usable outside of block creation or execution, all fields must be fully set upon creation without any zeroing.
 This means that the transaction ID must also include the correct `txPointer` value, not zeroed out.
 
-| name             | type                         | description                                          |
-|------------------|------------------------------|------------------------------------------------------|
-| `txPointer`      | [TXPointer](./tx-pointer.md) | The location of the `Mint` transaction in the block. |
-| `inputContract`  | [Input](./input.md)          | The contract utxo that assets are minted to.         |
-| `outputContract` | [Output](./output.md)        | The contract utxo that assets are being minted to.   |
-| `mintAmount`     | `uint64`                     | The amount of funds minted.                          |
-| `mintAssetId`    | `byte[32]`                   | The asset IDs corresponding to the minted amount.    |
+| name             | type                          | description                                          |
+|------------------|-------------------------------|------------------------------------------------------|
+| `txPointer`      | [TXPointer](./tx-pointer.md)  | The location of the `Mint` transaction in the block. |
+| `inputContract`  | [InputContract](./input.md)   | The contract utxo that assets are minted to.         |
+| `outputContract` | [OutputContract](./output.md) | The contract utxo that assets are being minted to.   |
+| `mintAmount`     | `uint64`                      | The amount of funds minted.                          |
+| `mintAssetId`    | `byte[32]`                    | The asset IDs corresponding to the minted amount.    |
 
 Transaction is invalid if:
 
 - `txPointer` is zero or doesn't match the block.
-- `inputContract` is not of the type `InputType.Contract`
-- `outputContract` is not of the type `OutputType.Contract`
+- `outputContract.inputIndex` is not zero
```

Comment:
> - `outputContract.inputIndex` is not zero

If I understand correctly, the mint transaction is effectively the coinbase. So it makes sense that the  has to be the first transaction. Is that what this change is saying?

> - `inputContract` is not of the type `InputType.Contract`
> - `outputContract` is not of the type `OutputType.Contract`

Are you removing these points because they are no longer true? Can the inputs and outputs be any type?

Reply:
> If I understand correctly, the mint transaction is effectively the coinbase. So it makes sense that the has to be the first transaction. Is that what this change is saying?

Nope, each `OutputContract` should be connected to `InputContract`. Because we always have only one `InputContract` the `OutputContract.inputIndex` can be and should be only `0`=)

> It looks like we are constraining the inputs and outputs to Contract, so you are removing these points because there is no other option.

Yep, instead of `Input` and `Output` we work with `InputContract` and `OutputContract` directly removing possibilities of other `Input`/`Output` variants.